app/client/src/components/RoleGuard.tsx

```diff
@@ -0,0 +1,14 @@
+import router from 'next/router';
+
+export interface RoleGuardProps {
+    organizer?: boolean;
+    moderator?: boolean;
+    children: JSX.Element | JSX.Element[];
+}
+
+export function RoleGuard({ organizer = false, moderator = false, children }: RoleGuardProps){
```

Comment:
Could you add a comment on what this component is for?  For example, `By default this component will not render any children unless the user is a moderator or an organizer of a particular event.`

Arguably, this component that should be using `useEvent` and checking whether or not the user is a moderator or an organizer.  Then, you could conditionally render children based on the parameters passed in by the caller. This would then be a more robust component.  What do you think?

Examples:
```tsx
// organizer only
<RoleGuard organizer>
...
</RoleGuard>

// moderator only (organizers have access because they are already a moderator)
<RoleGuard moderator>
...
</RoleGuard>
```